google/cloud/bigquery/dataset.py

```diff
@@ -241,7 +235,10 @@ def from_api_repr(cls, resource: dict) -> "AccessEntry":
         entity_type, entity_id = entry.popitem()
         if len(entry) != 0:
             raise ValueError("Entry has unexpected keys remaining.", entry)
-        return cls(role, entity_type, entity_id)
+
+        config = cls(role, entity_type, entity_id)
+        config._properties = copy.deepcopy(resource)
```

Comment:
@tswast does it make sense to do it this way? I can't tell if it's better to do something like
```
        entry = resource.copy()
        role = entry.pop("role", None)
        entity_type = entity_id = None       
        if entry:
            entity_type, entity_id = entry.popitem()
            if len(entry) != 0:
                raise ValueError("Entry has unexpected keys remaining.", entry)
        config = cls(role, entity_type, entity_id)
        config._properties = copy.deepcopy(resource)
        return config
  ```

Reply:
For some reason, fixing this issue in a way that anticipates the changes requested in #1077 is really throwing me off since I'm not familiar with this library.

Reply:
That works. I think it could be simpler than that and avoid the one of the two `deepcopy` calls by avoiding `pop` and `popitem`, but I'm okay leaving those as-is until we do the larger refactor.

Reply:
Note: we still need to update `to_api_repr` in this PR.

Reply:
Added